Documentation/ApiOverview/DependencyInjection/_ServicesWithInterfaceInjection.yaml

```diff
@@ -0,0 +1,21 @@
+services:
+  _defaults:
+    autowire: true
+    autoconfigure: true
+    public: false
+
+  # Define the default implementation of an interface
+  MyVendor\MyExtension\Domain\Repository\MyRepositoryInterface: '@MyVendor\MyExtension\Domain\Repository\MyDefaultRepository'
```

Comment:
That would mean, that every time you type hint MyRepositoryInterface the MyDefaultRepository is injected. What is a real use case for this (independent from "this can be done and works")?

Reply:
Well I just had that exact scenario: two controllers, one using default repository, one the other.

I think for documentation reasons it's not thaaaat important that the use case makes sense, but rather give an example of how the syntax of the Services.yaml looks like.

Reply:
As TYPO3 uses symfony/dependency-injection under the hood most of the features by this component are also possible in TYPO3 (like service subscribers). We do not want to copy the Symfony DI documentation. IMHO this chapter should be only an introduction to DI and hint to the specialties of TYPO3 (like GeneralUtility::makeInstance). Special cases, like aliasing interfaces and setting arguments can be documented here, when a specific use case is given which has value to TYPO3 and is not already documented in the Symfony documentation.

Your example now looks like it is a best practise which may be error-prone for newbies who do not understand DI already.

This is just my 2 cents, maybe someone else from the team has other opinions.

Reply:
Yup, but I have two problems with the pull request:

1. Changing a commonly-used LoggerInterface example (this is a best practise to inject an instance of a logger).
2. Use a custom Repository interface for this scenario (that is an edge case IMHO, for which cases do you have different repository classes). We should not showcase this as a best practise in the docs.

Reply:
I agree. So let me build on the logger example (which I prefered too!) and revert 0fdb42fbf750fd395a735ecf149b1e77e972f6c7 ?

Reply:
I would also prefer not to adjust the current example, but add an additional example and explain the use case.